mysys/my_bitmap.cc

```diff
@@ -136,37 +147,57 @@ static inline uint get_first_not_set(uint32 value, uint word_pos) {
   return MY_BIT_NONE;
 }
 
-bool bitmap_init(MY_BITMAP *map, my_bitmap_map *buf, uint n_bits) {
+bool bitmap_init(MY_BITMAP *map, my_bitmap_map *buf, uint n_bits,
+                 bool thread_safe) {
   DBUG_TRACE;
   if (!buf) {
     uint size_in_bytes = bitmap_buffer_size(n_bits);
     uint extra = 0;
 
+    if (thread_safe) {
+      size_in_bytes = ALIGN_SIZE(size_in_bytes);
```

Comment:
It's probably worth it to ensure that mutex will be located on a separate cache line

Reply:
I only restored the previously existing code here.